src/controllers/importer/ImporterController.php

```diff
@@ -0,0 +1,134 @@
+    return await $this->processTeams($teams);
+  }
+
+  public async function processLevels(array<string, mixed> $levels): Awaitable<bool> {
+    foreach ($levels as $level) {
+      $title = must_have_string($level, 'title');
+      $type = must_have_string($level, 'type');
+      $entity_name = must_have_string($level, 'entity_name');
+      $c = must_have_string($level, 'category');
+      $exist = await Level::genAlreadyExist($type, $title, $entity_name);
+      $entity_exist = await Country::genCheckExists($entity_name);
+      $category_exist = await Category::genCheckExists($c);
+      if (!$exist && $entity_exist && $category_exist) {
+        $entity = await Country::genCountry($entity_name);
+        $category = await Category::genSingleCategoryByName($c);
+        await Level::genCreate(
+          $type,
+          $title,
+          must_have_string($level, 'description'),
+          $entity->getId(),
+          $category->getId(),
+          must_have_int($level, 'points'),
+          must_have_int($level, 'bonus'),
+          must_have_int($level, 'bonus_dec'),
+          must_have_int($level, 'bonus_fix'),
+          must_have_string($level, 'flag'),
+          must_have_string($level, 'hint'),
+          must_have_int($level, 'penalty'),
+        );
+      }
+    }
+    return true;
+  }
+
+  public async function importLevels(string $input_filename): Awaitable<bool> {
+    $levels = $this->importData($input_filename, 'levels');
+    return await $this->processLevels($levels);
+  }
+
+  public async function processCategories(array<string, mixed> $categories): Awaitable<bool> {
+    foreach ($categories as $category) {
+      $c = must_have_string($category, 'category');
+      $exist = await Category::genCheckExists($c);
+      if (!$exist) {
+        await Category::genCreate(
+          $c,
+          (bool)must_have_idx($category, 'protected')
+        );
+      }
+    }
+    return true;
+  }
+
+  public async function importCategories(string $input_filename): Awaitable<bool> {
+    $categories = $this->importData($input_filename, 'categories');
+    return await $this->processCategories($categories);
+  }
+
+  public async function processLogos(array<string, mixed> $logos): Awaitable<bool> {
+    foreach ($logos as $logo) {
+      $name = must_have_string($logo, 'name');
+      $exist = await Logo::genCheckExists($name);
+      if (!$exist) {
+        await Logo::genCreate(
+          (bool)must_have_idx($logo, 'used'),
+          (bool)must_have_idx($logo, 'enabled'),
+          (bool)must_have_idx($logo, 'protected'),
+          $name,
+          must_have_string($logo, 'logo'),
+        );
+      }
+    }
+    return true;
+  }
+
+  public async function importLogos(string $input_filename): Awaitable<bool> {
+    $logos = $this->importData($input_filename, 'logos');
+    return await $this->processLogos($logos);
+  }
+
+  public async function importGame(string $input_filename): Awaitable<bool> {
```

Comment:
One problem I see with the way you're doing this function is that it's possible for part of the import to succeed, which might be bad. i.e. we could successfully process logos, and then fail to import teams.